src/users/users.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+
+@Injectable()
+export class UsersService {
+  private users = []; 
+
+  create(createUserDto: CreateUserDto) {
```

Comment:
please create interfaces and use it here
we don't need to use DTO here because data is allready valid, we validate it in controller
interfaces, unlike the DTOs, are not stored in final build, so it doesn't affect the runtime